elastic/apm-server
# Collect monitoring for the overflow metrics for all aggregations

## Motivation/summary

Collect monitoring for the overflow metrics for all aggregations. Some major changes in the PR are:
1. The PR moves the metric updates to `publish` method, this would mean that the metrics are updated periodically based on the metric interval (set to `1m`).
2. The overflow metrics collected represent an estimate of the total number of unique transaction groups that overflowed rather than total number of overflow.
3. Changes the metric name from `txmetrics.overflowed` to `txmetrics.overflowed.total`.
4. Introduces 3 more metrics: `txmetrics.overflowed.services`, `txmetrics.overflowed.per_service_txn_groups` and `txmetrics.overflowed.txn_groups` to represent overflow reason.

## Checklist

~~- [ ] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)~~
~~- [ ] Documentation has been updated~~

## How to test these changes

Create overflow as per [these steps](https://github.com/elastic/apm-server/issues/9648#issuecomment-1423607724) and check if the metrics are published.


## Related issues

Fixes #10207
